vendor/assets/javascripts/data-confirm-modal.js

```diff
@@ -137,6 +137,16 @@
       '</div>'
     );
 
+    // Make sure it's always the top zindex
+    var highest = 1050;
+    $('.modal.in').not('#'+id).each(function() {
+      current = parseInt($(this).css('z-index'), 10);
+      if(current > highest) {
+        highest = current
+      }
+      modal.css('z-index', parseInt(highest) + 1);
```

Comment:
This line should be outside of the `.each()` call, to avoid unnecessary `.css()` calls while iterating over the currently visible modals